packages/babel-parser/src/plugins/typescript/index.ts

```diff
@@ -4173,6 +4173,21 @@ export default (superClass: ClassWithMixin<typeof Parser, IJSXParserMixin>) =>
         );
       }
     }
+    checkIdentifier(
+      at: N.Identifier,
+      bindingType: BindingFlag,
+      strictModeChanged: boolean = false,
+    ) {
+      let oldStrict;
+      if (this.state.isAmbientContext && this.state.strict) {
+        oldStrict = this.state.strict;
```

Comment:
```suggestion
        oldStrict = true;
```

Reply:
Or can we change the strictness when entering the ambient context, rather than when checking the identifier?

Reply:
Or can we change the strictness when entering the ambient context, rather than when checking the identifier?